src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Semantics/Types/Type.cs

```diff
@@ -404,7 +366,7 @@ public ConstValKind constValKind()
         }
         public CType underlyingType()
         {
-            if (IsAggregateType() && getAggregate().IsEnum())
+            if (this is AggregateType && getAggregate().IsEnum())
```

Comment:
Anywhere that now has `this is SomeType` can be refactored further to avoid the test-and-dispatch pattern, but that can wait for later.